chaos_monkey/network/tcp_scanner.py

```diff
@@ -17,29 +16,25 @@ def is_host_alive(self, host):
         return self.get_host_fingerprint(host, True)
 
     def get_host_fingerprint(self, host, only_one_port=False):
-        assert isinstance(host, VictimHost)
+        """
+        Scans a target host to see if it's alive using the tcp_target_ports specified in the configuration.
+        :param host: VictimHost structure
+        :param only_one_port: Currently unused.
+        :return: T/F if there is at least one open port. In addition, the host object is updated to mark those services as alive.
+        """
 
-        count = 0
         # maybe hide under really bad detection systems
         target_ports = self._config.tcp_target_ports[:]
         shuffle(target_ports)
 
-        for target_port in target_ports:
-
-            is_open, banner = check_port_tcp(host.ip_addr,
-                                             target_port,
-                                             self._config.tcp_scan_timeout / 1000.0,
-                                             self._config.tcp_scan_get_banner)
-
-            if is_open:
-                count += 1
+        ports, banners = check_tcp_ports(host.ip_addr, target_ports, self._config.tcp_scan_timeout / 1000.0)
+        if len(ports) != 0:
```

Comment:
This if is unnecessary. Unsure if it helps readability. Keep it or remove it. your call.

Reply:
Without the if, just iterate and get null results? Ack